app/models/mixins/authentication_mixin.rb

```diff
@@ -266,9 +266,13 @@ def authentication_check(*args)
 
   private
 
+  def build_verify_args(type, options)
+    kind_of?(Host) ? [type, options] : type
+  end
+
   def authentication_check_no_validation(type, options)
     header  = "type: [#{type.inspect}] for [#{id}] [#{name}]"
-    verify_args = self.kind_of?(Host) ? [type, options] : type
+    verify_args = build_verify_args(type, options)
```

Comment:
np, I just opened a PR for that: https://github.com/ManageIQ/manageiq/pull/9872
I think it makes sense to have this refactoring in a different PR too